artemis-server/src/main/java/org/apache/activemq/artemis/core/config/impl/Validators.java

```diff
@@ -122,6 +122,30 @@ public void validate(final String name, final Object value) {
       }
    };
 
+   public static final Validator POSITIVE_INT = new Validator() {
+      @Override
+      public void validate(final String name, final Object value) {
+         Number val = (Number) value;
+         if (val.longValue() > 0 && val.longValue() <= Integer.MAX_VALUE) {
+            // OK
+         } else {
+            throw ActiveMQMessageBundle.BUNDLE.inRangeOfPositiveInt(name, val);
+         }
+      }
+   };
+
+   public static final Validator MINUS_ONE_OR_POSITIVE_INT = new Validator() {
```

Comment:
again not needed, if the xsd (schema is corrected)

Reply:
ignore i spotted, why finally, its because GB etc can be used in the string type, and then converted as such cannot limit this in xsd.